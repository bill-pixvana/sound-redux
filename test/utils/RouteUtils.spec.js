import * as RouteUtils from '../../client/utils/RouteUtils';


describe('RouteUtils', () => {
    describe('api', () => {
      it('should be correct', () => {expect(RouteUtils).toMatchSnapshot()});
    });
    describe('constructUrl', () => {
        it('should correctly construct a url', () => {
            const route = {path: ['songs'], query: {q: 'drake'}};
            expect(RouteUtils.constructUrl(route)).toEqual('songs?q=drake');
        });
    });

    describe('parseUrl', () => {
        it('should correctly parse a window hash', () => {
            expect(RouteUtils.parseUrl('songs?q=drake')).toEqual({path: ['songs'], query: {q: 'drake'}});
        });
    });
});
