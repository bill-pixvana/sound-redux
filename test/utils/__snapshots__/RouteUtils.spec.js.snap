// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteUtils api should be correct 1`] = `
Object {
  "constructUrl": [Function],
  "parseUrl": [Function],
}
`;
